ml_tools/eolearn/tests/test_train_split.py

```diff
@@ -36,58 +36,65 @@ def test_bad_args(bad_arg: Any, bad_kwargs: Any) -> None:
     for index, (i_1, i_2, j_1, j_2) in enumerate(indices, 1):
-        data[i_1:i_2, j_1:j_2, :] = index * 11
-
-    patch = EOPatch()
-    patch[INPUT_FEATURE] = data
+        eopatch1[INPUT_FEATURE][i_1:i_2, j_1:j_2, :] = index * 11
 
     bins = [0.2, 0.5, 0.8]
     expected_unique = set(range(1, len(bins) + 2))
 
-    patch = TrainTestSplitTask(INPUT_FEATURE, OUTPUT_FEATURE, bins, split_type=TrainTestSplitType.PER_CLASS)(
-        patch, seed=1
-    )
-    assert set(np.unique(patch[OUTPUT_FEATURE])) <= expected_unique
+    split_task = TrainTestSplitTask(INPUT_FEATURE, OUTPUT_FEATURE, bins, split_type=TrainTestSplitType.PER_CLASS)
+    eopatch1 = split_task(eopatch1, seed=1)
+    assert set(np.unique(eopatch1[OUTPUT_FEATURE])) <= expected_unique
 
-    result_seed1 = np.copy(patch[OUTPUT_FEATURE])
-    unique = (np.unique(result_seed1[i_1:i_2, j_1:j_2, :], return_counts=True) for i_1, i_2, j_1, j_2 in indices)
-    expected = [(i_2 - i_1) * (j_2 - j_1) * shape[-1] for i_1, i_2, j_1, j_2 in indices]
+    result = np.copy(eopatch1[OUTPUT_FEATURE])
+    unique = (np.unique(result[i_1:i_2, j_1:j_2, :], return_counts=True) for i_1, i_2, j_1, j_2 in indices)
+    expected = [(i_2 - i_1) * (j_2 - j_1) * eopatch1[OUTPUT_FEATURE].shape[-1] for i_1, i_2, j_1, j_2 in indices]
 
     for (unique_values, unique_counts), expected_count in zip(unique, expected):
         assert len(unique_values) == 1
         assert len(unique_counts) == 1
         assert unique_counts[0] == expected_count
 
-    # seed=2 should produce different result than seed=1
-    patch = TrainTestSplitTask(INPUT_FEATURE, OUTPUT_FEATURE, bins, split_type=TrainTestSplitType.PER_CLASS)(
-        patch, seed=2
-    )
-    result_seed2 = np.copy(patch[OUTPUT_FEATURE])
-    assert set(np.unique(result_seed2)) <= expected_unique
+
+def test_seed(eopatch1: EOPatch) -> None:
+    """seed=2 should produce different result than seed=1"""
+    bins = [0.2, 0.5, 0.8]
+    split_task = TrainTestSplitTask(INPUT_FEATURE, OUTPUT_FEATURE, bins, split_type=TrainTestSplitType.PER_CLASS)
+    result_seed1 = np.copy(split_task(eopatch1, seed=1)[OUTPUT_FEATURE])
+    result_seed2 = np.copy(split_task(eopatch1, seed=2)[OUTPUT_FEATURE])
+    result_seed1_rerun = np.copy(split_task(eopatch1, seed=1)[OUTPUT_FEATURE])
+
     assert not np.array_equal(result_seed1, result_seed2)
+    assert_array_equal(result_seed1, result_seed1_rerun)
 
-    # test with seed 1 should produce the same result as before
-    patch = TrainTestSplitTask(INPUT_FEATURE, OUTPUT_FEATURE, bins, split_type=TrainTestSplitType.PER_CLASS)(
-        patch, seed=1
-    )
-    result_seed_equal = patch[OUTPUT_FEATURE]
-    assert set(np.unique(result_seed2)) <= expected_unique
-    assert_array_equal(result_seed1, result_seed_equal)
 
-    # test ignore_values=[2]
+def test_ignore_value(eopatch1: EOPatch) -> None:
+    """test ignore_values=[2]"""
```

Comment:
this can be removed, the name of the test is informative enough

Reply:
done